src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addAnimal(animal) {
+    if (this.alive === undefined) {
+      this.alive = [];
+    }
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    this.alive = this.alive.filter(liveAnimal => animal !== liveAnimal);
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+    Animal.addAnimal(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.addAnimal(this);
+  }
+
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      Animal.removeAnimal(animal);
+    }
+  }
 }
 
```

Comment:
```suggestion
Animal.alive = [];
```
Because we changed how the array works, we can add this here